compiler/src/main/java/org/qbicc/type/definition/classfile/MethodParser.java

```diff
@@ -303,11 +303,13 @@ void setLocal2(int index, Value value, int bci) {
         if (value instanceof Auto al) {
             LocalVariableElement lve = getLocalVariableElement(bci, index);
             if (lve != null) {
-                Value allocated = gf.stackAllocate(lve.getType(), lf.literalOf(1), lf.literalOf(al.getPointeeType().getAlign()));
                 Value init = al.getInitializer();
-                if (init instanceof Literal lit && lit.isZero()) {
-                    init = lf.zeroInitializerLiteralOfType(lve.getType());
+                ValueType varType = init.getType();
+                if (init.getType().equals(ts.getVoidType())) {
+                    varType = lve.getType();
+                    init = lf.zeroInitializerLiteralOfType(varType);
                 }
+                Value allocated = gf.stackAllocate(varType, lf.literalOf(1), lf.literalOf(al.getPointeeType().getAlign()));
```

Comment:
I think the alignment probably should be `varType.getAlign()`, since `al`'s pointee type might be `void`.

Reply:
Changed.